src/imex.rs

```diff
@@ -472,7 +473,54 @@ async fn import_backup(context: &Context, backup_to_import: impl AsRef<Path>) ->
             .query_row(
                 "SELECT file_name, file_content FROM backup_blobs WHERE id = ?",
                 paramsv![file_id],
-                |row| Ok((row.get::<_, String>(0)?, row.get::<_, Vec<u8>>(1)?)),
+                // cs: This is the affected line NOW for producing the error. But
+                // here get is awaiting for String or another type to come back?
+                // |row| Ok((row.get::<_, String>(0)?, row.get::<_, Vec<u8>>(1)?)),
+
+                // cs: this is the secure replacement code for |row| ...
+                |row| {
+                    // Statement  'let name: String = row.get(0)?;'  is an assumption!
+                    //
+                    // This assumption that data matches String type is only true with clean utf8!
+                    // This simple "row.get(0) only works when TEXT column is ok and String conversion
+                    // from clean utf8 encoding is possible.
+                    // Every bad utf8 encoded data in TEXT column crashes this assumption and
+                    // switches to Err(_) path.
+                    let name: String = match row.get(0) {
+                        Ok(name) => name, // good utf8 filename
+                        Err(_) => {
+                            // bad utf8 encoding in filename
+                            match row.get_raw(0) {
+                                ValueRef::Null => "Null Value".to_string(),
```

Comment:
This sets filename to `Null Value`. Return an error with `bail!` instead, or make `name` an `Option`, set it to `None` and skip such files if you don't want to abort the import in this case.

Reply:
With this issue here I have a problem. A _continue_ in C would be a good solution but I don't know how to skip this one row.
IMHO I think it would make sense to skip this _row_and continue. A NULL Value column type never should appear in database except it is corrupt.

To get code compiled I keep the String return type as it is needed by the rest of the code.
Maybe You have a better proposal?

Reply:
This case is one where the file should be skipped. You cannot get a name at all.
What I'm doing here is to create a default name. But this approach is not covering every case!